index.html

```diff
@@ -566,6 +571,117 @@ <h1>Trust Model</h1>
+        </p>
+        <dl>
+          <dt><dfn>type</dfn></dt>
+          <dd>
+The value of this property MUST be one or more URIs.
+If more than one URI is provided, the array MUST be interpreted as a set of
+values where order is not significant. Note that syntax such as JSON-LD terms
+MAY be used to ease developer usage. It is RECOMMENDED that dereferencing the
+URI results in a document containing machine-readable information about the
+type.
+          </dd>
+        </dl>
+
+        <pre class="example nohighlight" title="Usage of the type property">
+{
+  "id": "http://dmv.example.gov/credentials/3732",
+  <span class="highlight">"type": ["VerifiableCredential", "ProofOfAgeCredential"]</span>,
+  "claim": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "ageOver": 21
+  },
+  "proof": { ... }
+}
+        </pre>
+
+        <p>
+With respect to this specification, the following <a>type</a>s MUST be
+specified on the following objects:
+        </p>
+
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>Concept</th>
+              <th>Type</th>
+            </tr>
+          </thead>
+
+          <tbody>
+            <tr>
+              <td>
+A <a href="#credentials">credential</a> object
+              </td>
+              <td>
+<tt>"type": ["<code>VerifiableCredential</code>"]</tt>
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+A <a href="#presentations">presentation</a> object
+              </td>
+              <td>
+<tt>"type": ["<code>VerifiablePresentation</code>"]</tt>
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+A <a href="#status">credentialStatus</a> object
+              </td>
+              <td>
+<tt>"type": ["<code>CredentialStatus</code>"]</tt> and
+a valid credential status type (e.g. <code>CredentialStatusList2017</code>)
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+A <a href="#terms-of-use">termsOfUse</a> object
+              </td>
+              <td>
+<tt>"type": ["<code>TermsOfUse</code>"]</tt> and
+a valid terms of use type (e.g. <code>OdrlPolicy2017</code>)
+              </td>
+            </tr>
+
+            <tr>
+              <td>
+An <a href="#terms-of-use">evidence</a> object
+              </td>
+              <td>
+<tt>"type": ["<code>Evidence</code>"]</tt> and
+a valid evidence type (e.g. <code>DocumentVerification2018</code>)
+              </td>
+            </tr>
+
+          </tbody>
+        </table>
+
+        <p class="issue" data-number="182">
+Whether or not <code>CredentialStatus</code>, <code>TermsOfUse</code>,
+or <code>Evidence</code> provide useful type information in addition to
+more specific types is currently a topic of active discussion.
+        </p>
+
+        <p>
+The type system for the Verifiable Credentials Data Model is the same as the
+one for [[JSON-LD]] and is covered in depth in that specification. All
```

Comment:
I don't find this statement acceptable for three reasons:
1. It has already been said that it is not necessary to know JSON-LD to work with VCs, but this statement is making it mandatory to fully understand JSON-LD in order to understand the use of the "type" property in VCs.
2. Having taken a quick look at the JSON-LD spec I could not find the "type" property described anywhere, although "@type" occurred a copious number of times. Are they the same or different?
3. The text for "@type" says specifically it is non-normative, viz

5.4 Specifying the Type
This section is non-normative.

The type of a particular node can be specified using the @type keyword. In Linked Data, types are uniquely identified with an IRI.

My conclusion is that the VC spec needs to definitively say what the 'type' property is and how the 'type' property is used

Reply:
> It has already been said that it is not necessary to know JSON-LD to work with VCs, but this statement is making it mandatory to fully understand JSON-LD in order to understand the use of the "type" property in VCs.

You don't need to have a full understanding of JSON-LD to use `type` in the same way as we don't require that developers have a full understanding of cryptography to use Verifiable Credentials. Most developers cut/paste code, which is made evident through the proper use of JSON-LD via schema.org on hundreds of millions of Web domains.

It's been suggested that we need to talk about `type` in this specification, which we borrow from JSON-LD and is defined in the JSON-LD specification (again, used on hundreds of millions of sites, so we do have deployment experience with it -- people seem to be able to use the feature correctly). We either duplicate the text in the JSON-LD specification in this specification (which is a bad specification writing practice), or we refer to it (which is considered the best practice currently). I suggest we refer to it. If we don't do that, we'll have to re-invent the type system just for Verifiable Credentials (which would effectively end up being JSON-LD's type system). We should re-use specs where we can.

To this end, I have referred to the exact section that needs to be read (both the non-normative and normative sections) in 8469ab0c3054c386c624c586db20ab71a9e96b04.

> Having taken a quick look at the JSON-LD spec I could not find the "type" property described anywhere, although "@type" occurred a copious number of times. Are they the same or different?

Clarified in 8469ab0c3054c386c624c586db20ab71a9e96b04.

> The text for "@type" says specifically it is non-normative, viz

Clarified in 8469ab0c3054c386c624c586db20ab71a9e96b04.